File: frontend-hormonia/src/lib/api-client/hive-mind.ts
import { ApiClientCore } from './core'

// ============================================================================
// HIVE MIND TYPES
// ============================================================================

export interface AgentStatus {
  agent_id: string
  agent_type: string
  status: string
  capabilities: string[]
  health?: {
    is_healthy: boolean
    response_time: number
    success_rate: number
    active_tasks: number
    error_count: number
  }
}

export interface SystemHealthOverview {
  status: 'healthy' | 'degraded' | 'down'
  timestamp: string
  system_health: {
    total_agents: number
    active_agents: number
    system_uptime_seconds: number
    agents_by_status: Record<string, number>
    active_system_alerts: number
    total_active_alerts: number
  }
  system_alerts: number
}

// ============================================================================
// HIVE MIND API
// ============================================================================

export interface HiveMindApi {
  health: () => Promise<SystemHealthOverview>
  agents: {
    list: () => Promise<{ agents: AgentStatus[]; total_agents: number; timestamp: string }>
  }
}

export function createHiveMindApi(client: ApiClientCore): HiveMindApi {
  return {
    health: () => client.get('/api/v2/hive-mind/health'),

    agents: {
      list: () => client.get('/api/v2/hive-mind/agents'),
    },
  }
}

File: backend-hormonia/app/schemas/v2/hive_mind.py
from datetime import datetime
from typing import List, Optional, Dict, Any
from pydantic import BaseModel

class AgentHealthSchema(BaseModel):
    is_healthy: bool
    response_time: float
    success_rate: float
    active_tasks: int
    error_count: int

class AgentStatusSchema(BaseModel):
    agent_id: str
    agent_type: str
    status: str
    specialization: Optional[str] = None
    capabilities: List[str]
    health: Optional[AgentHealthSchema] = None

class AgentListResponse(BaseModel):
    agents: List[AgentStatusSchema]
    total_agents: int
    timestamp: datetime

class SystemHealthOverview(BaseModel):
    status: str
    timestamp: datetime
    system_health: Dict[str, Any]
    system_alerts: int
